packages/plugin-page-url-enrichment-browser/src/page-url-enrichment.ts

```diff
@@ -0,0 +1,209 @@
+  Internal = 'internal', // for same domain - this excludes subdomains
+  External = 'external', // for different domains
+}
+
+export const isPageUrlEnrichmentEnabled = (option: unknown): boolean => {
+  if (typeof option === 'boolean') {
+    return option;
+  }
+  if (typeof option === 'object' && option !== null && 'pageUrlEnrichment' in option) {
+    return Boolean((option as { pageUrlEnrichment?: boolean }).pageUrlEnrichment);
+  }
+  return false;
+};
+
+export const pageUrlEnrichmentPlugin = (): EnrichmentPlugin => {
+  const globalScope = getGlobalScope();
+  let sessionStorage: BrowserStorage<URLInfo> | undefined = undefined;
+  let isStorageEnabled = false;
+  let loggerProvider: Logger | undefined = undefined;
+
+  let isProxied = false;
+  let isTracking = false;
+
+  const getDecodeURI = (locationStr: string): string => {
+    let decodedLocationStr = locationStr;
+    try {
+      decodedLocationStr = decodeURI(locationStr);
+    } catch (e) {
+      /* istanbul ignore next */
+      loggerProvider?.error('Malformed URI sequence: ', e);
+    }
+
+    return decodedLocationStr;
+  };
+
+  const getHostname = (url: string): string | undefined => {
+    let hostname: string | undefined;
+
+    try {
+      const decodedUrl = getDecodeURI(url);
+      hostname = new URL(decodedUrl).hostname;
+    } catch (e) {
+      /* istanbul ignore next */
+      loggerProvider?.error('Could not parse URL: ', e);
+    }
+
+    return hostname;
+  };
+
+  const getPrevPageType = (previousPage: string) => {
+    const currentDomain = (typeof location !== 'undefined' && location.hostname) || '';
+    const previousPageDomain = previousPage ? getHostname(previousPage) : undefined;
+
+    switch (previousPageDomain) {
+      case undefined:
+        return PreviousPageType.Direct;
+      case currentDomain:
+        return PreviousPageType.Internal;
+      default:
+        return PreviousPageType.External;
+    }
+  };
+
+  const saveURLInfo = async () => {
+    if (sessionStorage && isStorageEnabled) {
+      const URLInfo = await sessionStorage.get(URL_INFO_STORAGE_KEY);
+      const previousURL = URLInfo?.[CURRENT_PAGE_STORAGE_KEY] || '';
+      const currentURL = getDecodeURI((typeof location !== 'undefined' && location.href) || '');
+
+      await sessionStorage.set(URL_INFO_STORAGE_KEY, {
+        [CURRENT_PAGE_STORAGE_KEY]: currentURL,
+        [PREVIOUS_PAGE_STORAGE_KEY]: previousURL,
+      });
+    }
+  };
+
+  const saveUrlInfoWrapper = () => {
+    void saveURLInfo();
+  };
+
+  const plugin: EnrichmentPlugin = {
+    name: '@amplitude/plugin-page-url-enrichment-browser',
+    type: 'enrichment',
+
+    setup: async (config: BrowserConfig, _: BrowserClient) => {
+      loggerProvider = config.loggerProvider;
+      loggerProvider.log('Installing @amplitude/plugin-page-url-enrichment-browser');
+
+      isTracking = true;
+
+      if (globalScope) {
+        sessionStorage = new BrowserStorage<URLInfo>(globalScope.sessionStorage);
```

Comment:
from a product perspective, do we want to get access to previous page url in this case where a new page opens through a button click? If so, session storage won't meet the needs here as it's not shared between tabs.